### PR TITLE
add run line mark for local run button of functions

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import java.util.Iterator;
+
+public class FunctionRunConfigurationProducer extends LazyRunConfigurationProducer<AzureRunConfigurationBase> {
+    @NotNull
+    @Override
+    public ConfigurationFactory getConfigurationFactory() {
+        return AzureFunctionSupportConfigurationType.getInstance().getConfigurationFactories()[0];
+    }
+
+    public static boolean isFunctionClassAnnotated(final PsiMethod method) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java]: Move this method to `FunctionUtils`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        final Location contextLocation = context.getLocation();
+        assert contextLocation != null;
+        Location<PsiMethod> methodLocation = getAzureFunctionMethods(contextLocation);
+        if (methodLocation == null) {
+            return false;
+        }
+        final RunnerAndConfigurationSettings template = context.getRunManager().getConfigurationTemplate(getConfigurationFactory());
+
+        final Module contextModule = context.getModule();
+        final Module predefinedModule = ((FunctionRunConfiguration) template.getConfiguration()).getConfigurationModule().getModule();
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java]: Can we use configuration here directly?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunLineMarkerProvider.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    @Nullable
+    @Override
+    public Info getInfo(@NotNull PsiElement e) {
+        if (isIdentifier(e)) {
+            PsiElement element = e.getParent();
+            if (element instanceof PsiMethod &&
+                            FunctionRunConfigurationProducer.isFunctionClassAnnotated((PsiMethod) element)) {
+                final AnAction[] actions = ExecutorAction.getActions();
+                return new Info(AllIcons.RunConfigurations.TestState.Run, actions,
+                        element1 -> StringUtil.join(ContainerUtil.mapNotNull(actions, action -> getText(action, element1)),
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunLineMarkerProvider.java]: rename to `psiElement`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java (hunk 2 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java]: For config run from runline bar, there will be no chance for viewing and editing debug settings, use initializeDefaults to set null values like staging folder to a reasonable default value.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java (hunk 1 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunConfigurationProducer.java]: Done

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunLineMarkerProvider.java (hunk 3 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/FunctionRunLineMarkerProvider.java]: done

**Pull request merged**